sdk/identity/identity/src/msal/nodeFlows/msalNodeCommon.ts

```diff
@@ -175,6 +184,37 @@ export abstract class MsalNode extends MsalBaseUtilities implements MsalFlow {
     };
   }
 
+  protected getApp(
```

Comment:
the app types are fixed? maybe you can add overloads, then there is no need to cast the result of `getApp()`

https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8BvAvgKFJKAhBLWoGMB7AOwGdgoBDMMABQFcAjAGwEsCAuWKAHygYkAJhABmbEhCEBuQqQrVaAYQBOUiCWBsqLbnH6CR4yTPyjBBLaSgBzCMBi0AFBgjcARGGbsCAMTYqFO58UO7EJOIimtos-oHA7gCUeiFYsuYklmzWdg7Orh5erBxJeukWViS29o5gLuBuoeGRGlo6pdjlmZXVeXUFoUU+cUEhYaQt0TojCWNDJWPNbFFtLB362FCIqFC7UGyiUPWY8KeD3hwzSVs7e3dqwAwqVU409BcEfPxvqupTLIkAIS3XbIKAQFhkaAHI6uBBncYRZatGJXRI3O73exPF4-NQrGJfRTvYoEIEgqBgiFQ-aHY7QU7wJoTZH-a7bTF7B444m-Ak6QGyO5UyHQDmcqDc57ExikwUUtBoOTkShqMgMFiUJm5WpOTwfJLSIA

Reply:
publicFirst / credentialFirst still return union types thus requires casting.  Please discard.

Reply:
done